source/tools/AutomatedTestSuite.h

```diff
@@ -0,0 +1,262 @@
+/*
+ * Copyright (C) 2016 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef SYNTHMARK_AUTOMATED_TEST_SUITE_H
+#define SYNTHMARK_AUTOMATED_TEST_SUITE_H
+
+#include <cmath>
+#include <cstdint>
+
+#include "tools/LatencyMarkHarness.h"
+#include "tools/TimingAnalyzer.h"
+#include "tools/UtilizationMarkHarness.h"
+#include "tools/VirtualAudioSink.h"
+#include "tools/VoiceMarkHarness.h"
+
+constexpr double kHighLowThreshold = 0.1; // Difference between CPUs to consider them BIG-little.
+constexpr double kMaxUtilization = 0.9; // Maximum load that we will push the CPU to.
+
+/**
+ * Run an automated test, analyze the results and print a report.
+ * 1) Determine what CPUs are available
+ * 2) Run VoiceMark on each CPU to determine big vs little CPUs
```

Comment:
For the sake of discussion, what about using something like
`cat /proc/cpuinfo | grep mips`
?

Reply:
Being picky, is it OK to assume that the two device categories are SMP and big.LITTLE?
I mean, what if for some reason we have a device with different frequencies for each processor?
Designing for the future, do you think it would be valuable to generalize this approach for "more heterogeneous" architectures?

Reply:
cat /proc/cpuinfo does not work on some systems, eg. Mac OSX.

The code is not the ultimate code. It is just more than we had in the past. Ultimately, yes, we should run tests on every CPU in case of completely heterogeneous architectures.